dietpi/dietpi-software

```diff
@@ -12753,6 +12683,26 @@ _EOF_
 		if To_Uninstall 182 # Unbound
 		then
 			# Pi-hole: Assure that it does not resolve via Unbound anymore
+			# - v6
+			if [[ $(pihole-FTL -v) == 'v6'* ]]
+			then
+				G_DIETPI-NOTIFY 2 'Removing Unbound from Pi-hole upstream DNS servers, and in case add Quad9 as new one.'
+				local upstreams i
+				read -ra upstreams < <(pihole-FTL --config dns.upstreams)
+				# Remove array brackets
+				unset -v 'upstreams[0]' 'upstreams[${#upstreams[@]}]'
+				for i in "${!upstreams[@]}"
+				do
+					# Remove loopback IP (Unbound)
+					[[ ${upstreams[i]} == '127.0.0.1'* ]] && unset -v 'upstreams[i]'
+					# Remove trailing comma and readd double-quotes removed from pihole-FTL output
+					upstreams[i]="\"${upstreams[i]%,}\""
+				done
+				# Add Quad9 if no upstream DNS server is left
```

Comment:
somehow this is not working. At least it did not add Quad9 on Unbound removal

```
[ INFO ] DietPi-Software | Removing Unbound from Pi-hole upstream DNS servers, and in case add Quad9 as new one.
[  OK  ] DietPi-Software | pihole-FTL,--config,dns.upstreams,[ "" ]
[ INFO ] DietPi-Software | APT purge unbound, please wait...
```

Reply:
The `$IPS` did apply for `G_EXEC` already, having arguments internally comma-separated instead of space-separated. That was intended for expanding the array only. Works without `G_EXEC`, but I solved it now differently, converting the array to a string first, then replacing the spaces with comma separately, without `$IPS`.

Reply:
ok but it did not set the Quad9 upstream DNS server on Unbound removal

```
[  OK  ] DietPi-Software | pihole-FTL --config dns.upstreams [ "" ]
```

Reply:
Ah, yeah I just ran into the same and fixed it.